packages/eslint-plugin/tests/rules/no-extra-parens.test.ts

```diff
@@ -242,6 +242,11 @@ f<(number | string)[]>(['a', 1])
         },
       },
     }),
+    {
+      code: `
+const a = (['a', 'b'] satisfies A[]).includes('a');
```

Comment:
Recreating a new thread from https://github.com/typescript-eslint/typescript-eslint/pull/7026/files#r1192868544: shouldn't this still cause a rule report?

```ts
const a = ((['a', 'b'] satisfies A[])).includes('a');
```

Note the extra parenthesis around the `(['a', 'b'] satisfies A[])`.

Reply:
sure, this case definitely deserves a test